src/compiler/binder.ts

```diff
@@ -424,6 +425,7 @@ module ts {
                 case SyntaxKind.FunctionDeclaration:
                 case SyntaxKind.FunctionExpression:
                 case SyntaxKind.ArrowFunction:
+                case SyntaxKind.TypeAliasDeclaration:

```

Comment:
Why does this get its type parameters added to `.locals`, but Class and Interface do not? Are the type parameters added to the locals for generic signatures as well?